pkg/server/router/credential_test.go

```diff
@@ -194,4 +201,97 @@ func TestCredentialRouter(t *testing.T) {
 		assert.Error(tt, err)
 		assert.Contains(tt, err.Error(), fmt.Sprintf("credential not found with id: %s", cred.ID))
 	})
+
+	t.Run("Credential Status List Test", func(tt *testing.T) {
```

Comment:
I typically advocate for writing unit tests that test a single behavior. This is also known as [avoiding multiple acts](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#avoid-multiple-acts). In this case, I'd recommend separating this into multiple different behaviors that you're testing.

I know this isn't the style that we've been following for tests, so I'm ok to leave for consistency; but I do think it makes for tests that are easier to read, and thus maintain.

Reply:
agree with this our tests need some improvement. it could be good to start the practice now (and going forward) and open up some issues for refactoring our existing tests.

Reply:
I opened #173